lib/spack/spack/test/bootstrap.py

```diff
@@ -5,10 +5,19 @@
 import pytest
 
 import spack.bootstrap
+import spack.environment
 import spack.store
 import spack.util.path
 
 
+@pytest.fixture
+def active_mock_environment(mutable_config, mutable_mock_env_path):
+    env = spack.environment.create('bootstrap-test')
+    spack.environment.activate(env)
+    yield env
+    spack.environment.deactivate()
```

Comment:
Environment has `__enter__` and `__exit___`, so you don't have to introduce another fixture :thinking: 

```
env = spack.environment.create(...)
with env:
  with spack.bootstrap.ensure_bootstrap_configuration():
    assert ...
  asset ...
```

Reply:
I also agree with using `env` as a context manager, but it suffers from the same shell issue as above. In both cases we would assume `shell=sh`. Also I would slightly prefere leaving the fixture in any case - setting up and activating the environment is a prerequisite to the test.

Reply:
So, current state right now: using the context manager and a fixture.